addons/sale/models/sale_order.py

```diff
@@ -941,7 +941,7 @@ def action_update_taxes(self):
     def action_update_prices(self):
         self.ensure_one()
 
-        self._recompute_prices()
+        self.with_context(update_prices_button=True)._recompute_prices()
```

Comment:
If the only use of this context key is in `sale_subscription`, you could specify it there with a dedicated override, without having to modify the `sale` module ;).
```python
def action_update_prices(self):
    return super(SaleOrder, self.with_context(update_prices_button=True)).action_update_prices()
```